sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -190,25 +185,23 @@ pi_result OCL(piProgramCreate)(pi_context context, const void *il,
 
   std::vector<cl_device_id> devicesInCtx(deviceCount);
 
-  ret_err = clGetContextInfo(cast<cl_context>(context), CL_CONTEXT_DEVICES,
-                             deviceCount * sizeof(cl_device_id),
-                             devicesInCtx.data(), NULL);
-
   if (ret_err != CL_SUCCESS || deviceCount < 1) {
     if (res_program != nullptr)
       *res_program = nullptr;
     return cast<pi_result>(CL_INVALID_CONTEXT);
   }
 
+  ret_err = clGetContextInfo(cast<cl_context>(context), CL_CONTEXT_DEVICES,
+                             deviceCount * sizeof(cl_device_id),
+                             devicesInCtx.data(), NULL);
```

Comment:
```suggestion
                             devicesInCtx.data(), nullptr);
```

Reply:
The field is size_t type.
http://khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetContextInfo.html

Reply:
It looks like a mistake on the webpage i gave. I'll correct the code to use nullptr.